From 69050488368362821192b7f0a670abe3bed11c00 Mon Sep 17 00:00:00 2001
From: Zach Reizner <zachr@google.com>
Date: Fri, 3 Oct 2014 16:21:28 -0700
Subject: [PATCH 2/3] Revert "intel: restrict dma-buf-import images to external
 sampling only"

This reverts commit 3a52cd351af6ad834fa6205ec03a82559d1cea92.

Conflicts:
	src/mesa/drivers/dri/i965/intel_image.h
---
 src/mesa/drivers/dri/i965/intel_fbo.c    | 7 -------
 src/mesa/drivers/dri/i965/intel_image.h  | 9 +--------
 src/mesa/drivers/dri/i965/intel_screen.c | 1 -
 3 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/src/mesa/drivers/dri/i965/intel_fbo.c b/src/mesa/drivers/dri/i965/intel_fbo.c
index 4a03b57..4a04528 100644
--- a/src/mesa/drivers/dri/i965/intel_fbo.c
+++ b/src/mesa/drivers/dri/i965/intel_fbo.c
@@ -363,13 +363,6 @@ intel_image_target_renderbuffer_storage(struct gl_context *ctx,
       return;
    }
 
-   /* Buffers originating from outside are for read-only. */
-   if (image->dma_buf_imported) {
-      _mesa_error(ctx, GL_INVALID_OPERATION,
-            "glEGLImageTargetRenderbufferStorage(dma buffers are read-only)");
-      return;
-   }
-
    /* __DRIimage is opaque to the core so it has to be checked here */
    switch (image->format) {
    case MESA_FORMAT_R8G8B8A8_UNORM:
diff --git a/src/mesa/drivers/dri/i965/intel_image.h b/src/mesa/drivers/dri/i965/intel_image.h
index 0cae711..e6ad06e 100644
--- a/src/mesa/drivers/dri/i965/intel_image.h
+++ b/src/mesa/drivers/dri/i965/intel_image.h
@@ -89,14 +89,7 @@ struct __DRIimageRec {
    GLuint tile_y;
    bool has_depthstencil;
 
-   /**
-    * Provided by EGL_EXT_image_dma_buf_import.
-    *
-    * The flag is set in order to restrict the use of the image later on.
-    *
-    * See intel_image_target_texture_2d()
-    */
-   bool dma_buf_imported;
+   /* Provided by EGL_EXT_image_dma_buf_import */
    enum __DRIYUVColorSpace yuv_color_space;
    enum __DRISampleRange sample_range;
    enum __DRIChromaSiting horizontal_siting;
diff --git a/src/mesa/drivers/dri/i965/intel_screen.c b/src/mesa/drivers/dri/i965/intel_screen.c
index 88b5b13..6f92052 100644
--- a/src/mesa/drivers/dri/i965/intel_screen.c
+++ b/src/mesa/drivers/dri/i965/intel_screen.c
@@ -732,7 +732,6 @@ intel_create_image_from_dma_bufs(__DRIscreen *screen,
       return NULL;
    }
 
-   image->dma_buf_imported = true;
    image->yuv_color_space = yuv_color_space;
    image->sample_range = sample_range;
    image->horizontal_siting = horizontal_siting;
-- 
1.9.1

